ext/session/session.c

```diff
@@ -355,34 +355,28 @@ PHPAPI zend_string *php_session_create_id(PS_CREATE_SID_ARGS) /* {{{ */
 /* Default session id char validation function allowed by ps_modules.
  * If you change the logic here, please also update the error message in
  * ps_modules appropriately */
-PHPAPI zend_result php_session_valid_key(const char *key) /* {{{ */
+PHPAPI zend_result php_session_valid_key(const zend_string *str) /* {{{ */
 {
-	size_t len;
-	const char *p;
-	char c;
-	zend_result ret = SUCCESS;
+	const char *key = ZSTR_VAL(str);
 
-	for (p = key; (c = *p); p++) {
+	for (size_t i = 0; i < str->len; i++) {
```

Comment:
You should not access `len` directly, please use `ZSTR_LEN()`.

Reply:
Thanks, changed.